src/Core/CacheManager.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Core
+{
+    public class CacheManager
+    {
+        private readonly Func<Task<IReadOnlyDictionary<string, IReadOnlyDictionary<string, string>>>> _asyncDataProvider;
+
+        private IReadOnlyDictionary<string, IReadOnlyDictionary<string, string>> _data;
+
+        private IReadOnlyDictionary<string, IReadOnlyDictionary<string, string>> GetData()
+        {
+            if (_data != null) return _data;
+
+            _data = _asyncDataProvider().Result;
```

Comment:
Next step, do this async.
How to Handle startup. https://andrewlock.net/running-async-tasks-on-app-startup-in-asp-net-core-part-1/
Kika med Eric om alternativ och tankar från nuvarande Localizer i Assets.